openeo/extra/job_management.py

```diff
@@ -112,6 +104,20 @@ def get_by_status(self, statuses: List[str], max=None) -> pd.DataFrame:
         """
         ...
 
+    @abc.abstractmethod
+    def initialize_from_df(self, df: pd.DataFrame, on_exists: str = "error") -> "JobDatabaseInterface":
```

Comment:
I'm not sure this method must be  a required part of  the  `JobDatabaseInterface` interface. There are still some conceptual issues with `initialize_from_df` at the moment (e.g. see #667), so we might want to avoid painting ourselves in the corner here.

 `initialize_from_df` is not something that should be called automatically from within the job manager. It is explicitly intended for end users to call explicitly on their job database. So in that sense, it should not be part of the  `JobDatabaseInterface` interface, which is a contract between the developer of the database  and job manager, not a contract between user and job manager.